### PR TITLE
CDD-1814: Added X-Submitter-Identifier header.

### PR DESCRIPTION
**Pull request by rashadmughal**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthActionCustomHeader.scala [hunk 1]
```diff
@@ -17,34 +17,59 @@
 package uk.gov.hmrc.customs.declaration.controllers.actionbuilders
 
 import javax.inject.{Inject, Singleton}
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthActionCustomHeader.scala]: unused import `import play.api.mvc.Result`

File: app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthActionCustomHeader.scala [hunk 2]
```diff
@@ -17,34 +17,59 @@
...
+import play.api.mvc.Result
 import uk.gov.hmrc.customs.api.common.controllers.ErrorResponse
 import uk.gov.hmrc.customs.declaration.connectors.GoogleAnalyticsConnector
 import uk.gov.hmrc.customs.declaration.controllers.CustomHeaderNames._
 import uk.gov.hmrc.customs.declaration.logging.DeclarationsLogger
 import uk.gov.hmrc.customs.declaration.model._
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthActionCustomHeader.scala]: unused import `import uk.gov.hmrc.customs.declaration.model.actionbuilders.ActionBuilderModelHelper._
`

File: app/uk/gov/hmrc/customs/declaration/logging/LoggingHelper.scala [hunk 3]
```diff
@@ -61,7 +61,7 @@ object LoggingHelper {
         a.authorisedAs match {
           case Csp(badgeIdentifier, _) => s"[authorisedAs=Csp($badgeIdentifier)]"
           case NonCsp(eori, _) => s"[authorisedAs=NonCsp($eori)]"
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/declaration/logging/LoggingHelper.scala]: `authorisedAs=FileUploadCsp` looks wrong - this class has been renamed to `CspWithEori` (more generic)

File: app/uk/gov/hmrc/customs/declaration/xml/MdgPayloadDecorator.scala [hunk 4]
```diff
@@ -35,11 +35,18 @@ class MdgPayloadDecorator() {
...
-        { vpr.authorisedAs match {
+        {val as = vpr.authorisedAs
+      as match {
             case Csp(badgeId, _) => <v1:badgeIdentifier>{badgeId.value}</v1:badgeIdentifier>
-            case _ => NodeSeq.Empty
+            case NonCsp(eori, _) =>
+              <v1:authenticatedPartyID>{eori.value}</v1:authenticatedPartyID> // originatingPartyID is only required for CSPs
+            case CspWithEori(badgeId, eori, _) =>
+              <v1:badgeIdentifier>{badgeId.value}</v1:badgeIdentifier>
+              <v1:originatingPartyID>{eori.value}</v1:originatingPartyID>
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/declaration/xml/MdgPayloadDecorator.scala]: question - are we sure `asfr.fields.authenticatedEori` will exists?

File: test/component/CustomsDeclarationSubmissionSpec.scala [hunk 5]
```diff
@@ -97,6 +97,7 @@ class CustomsDeclarationSubmissionSpec extends ComponentTestSpec with AuditServi
       status(result) shouldBe ACCEPTED
```
Comment (googley42) [inline test/component/CustomsDeclarationSubmissionSpec.scala]: I think for the 3 services, in the component tests we need a call to `MdgWcoDecService.verifyMdgWcoDecServiceWasCalledWith` to verify the payload contains the new fields

File: app/uk/gov/hmrc/customs/declaration/xml/MdgPayloadDecorator.scala (hunk 4 above)
Comment (rashadmughal) [reply app/uk/gov/hmrc/customs/declaration/xml/MdgPayloadDecorator.scala]: yes - I check it before hand: https://github.com/hmrc/customs-declarations/pull/203/files#diff-4ec69ceb56f18e0df51ecc2ec0655524R135

File: app/uk/gov/hmrc/customs/declaration/controllers/CustomHeaderNames.scala [hunk 6]
```diff
@@ -24,4 +24,5 @@ object CustomHeaderNames {
   val NonRepudiationReceiptId: String = "X-Receipt-ID"
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/declaration/controllers/CustomHeaderNames.scala]: please updated `README.md` with mention of the new api-subscription-field and include it in the the example curl statement `curl -v -X PUT "http://localhost:9650/field/application/d65f2252-9fcf-4f04-9445-5971021226bb/context/customs%2Fdeclarations/version/2.0" -H "Cache-Control: no-cache" -H "Content-Type: application/json" -d '{ "fields" : { "callbackUrl" : "https://postman-echo.com/post", "securityToken" : "securityToken" } }'`

**Pull request merged**